quickstart/docker/image/ziti-cli-functions.sh

```diff
@@ -62,30 +64,46 @@ function initializeController {
 function startZitiController {
   log_file="${ZITI_HOME-}/${ZITI_EDGE_CONTROLLER_RAWNAME}.log"
   # shellcheck disable=SC2034
-  ("${ZITI_BIN_DIR-}/ziti-controller" run "${ZITI_HOME_OS_SPECIFIC}/${ZITI_EDGE_CONTROLLER_RAWNAME}.yaml" > "${log_file}" 2>&1 &)
-  pid=$!
-  echo -e "ziti-controller started as process id: $pid. log located at: $(BLUE "${log_file}")"
-  return $pid
+  "${ZITI_BIN_DIR-}/ziti-controller" run "${ZITI_HOME_OS_SPECIFIC}/${ZITI_EDGE_CONTROLLER_RAWNAME}.yaml" &> "${log_file}" &
+  ZITI_EXPRESS_CONTROLLER_PID=$!
+  echo -e "ziti-controller started as process id: $ZITI_EXPRESS_CONTROLLER_PID. log located at: $(BLUE "${log_file}")"
 }
 
 function stopZitiController {
-  killall ziti-controller
-  # shellcheck disable=SC2181
-  if [[ $? == 0 ]]; then echo "Controller stopped."; fi
+  if [[ -n ${ZITI_EXPRESS_CONTROLLER_PID:-} ]]; then
+    kill "$ZITI_EXPRESS_CONTROLLER_PID"
+    # shellcheck disable=SC2181
+    if [[ $? == 0 ]]; then 
+      echo "Controller stopped."
+      return 0
+    fi
+  else
+    echo "ERROR: you can only stop a controller process that was started with startZitiController" >&2
+    return 1
+  fi
 }
 
 function startExpressEdgeRouter {
   log_file="${ZITI_HOME_OS_SPECIFIC}/${ZITI_EDGE_ROUTER_RAWNAME}.log"
   "${ZITI_BIN_DIR}/ziti-router" run "${ZITI_HOME_OS_SPECIFIC}/${ZITI_EDGE_ROUTER_RAWNAME}.yaml" > "${log_file}" 2>&1 &
-  pid=$!
-  echo -e "Express Edge Router started as process id: $pid. log located at: $(BLUE "${log_file}")"
-  return $pid
+  ZITI_EXPRESS_EDGE_ROUTER_PID=$!
+  echo -e "Express Edge Router started as process id: $ZITI_EXPRESS_EDGE_ROUTER_PID. log located at: $(BLUE "${log_file}")"
+  
 }
 
 function stopAllEdgeRouters {
-  killall ziti-router
-  # shellcheck disable=SC2181
-  if [[ $? == 0 ]]; then echo "Router(s) stopped."; fi
+  if [[ -n ${ZITI_EXPRESS_EDGE_ROUTER_PID:-} ]]; then
+    # shellcheck disable=SC2015
+    kill "${ZITI_EXPRESS_EDGE_ROUTER_PID}" && {
+      echo "INFO: stopped router"
+    } || {
+      echo "ERROR: something went wrong with stopping the router(s)" >&2
+      return 1
+    }
+  else
+    echo "ERROR: you can only stop a router process that was started with startExpressEdgeRouter" >&2
```

Comment:
probably should rename this function now to `stopExpressEdgeRouter`. also, that name is inconsistent and it's bugged me for a while. Maybe also change `startExpressEdgeRouter` to `startEdgeRouter` ? or ... i dunno something to clean/true these ideas up?

Reply:
Good idea. I trued up the start/stop function names.